packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart

```diff
@@ -12,245 +15,292 @@ import 'package:flutter_tools/src/cache.dart';
+    fileSystem.file(fileSystem.path.join(projectPath, 'pubspec.yaml'))
+        ..createSync(recursive: true)
+        ..writeAsStringSync(pubspecYamlSrc);
+    _createDotPackages(projectPath);
+  });
 
-      // Analyze in the current directory - no arguments
-      testUsingContext('working directory', () async {
-        await runCommand(
-          command: AnalyzeCommand(workingDirectory: globals.fs.directory(projectPath)),
-          arguments: <String>['analyze', '--no-pub'],
-          statusTextContains: <String>['No issues found!'],
-        );
-      });
-
-      // Analyze a specific file outside the current directory
-      testUsingContext('passing one file throws', () async {
-        await runCommand(
-          command: AnalyzeCommand(),
-          arguments: <String>['analyze', '--no-pub', libMain.path],
-          toolExit: true,
-          exitMessageContains: 'is not a directory',
-        );
-      });
+  setUp(() {
+    libMain = fileSystem.file(fileSystem.path.join(projectPath, 'lib', 'main.dart'))
+        ..createSync(recursive: true)
+        ..writeAsStringSync(mainDartSrc);
+  });
 
-      // Analyze in the current directory - no arguments
-      testUsingContext('working directory with errors', () async {
-        // Break the code to produce the "Avoid empty else" hint
-        // that is upgraded to a warning in package:flutter/analysis_options_user.yaml
-        // to assert that we are using the default Flutter analysis options.
-        // Also insert a statement that should not trigger a lint here
-        // but will trigger a lint later on when an analysis_options.yaml is added.
-        String source = await libMain.readAsString();
-        source = source.replaceFirst(
-          'return MaterialApp(',
-          'if (debugPrintRebuildDirtyWidgets) {} else ; return MaterialApp(',
-        );
-        source = source.replaceFirst(
-          'onPressed: _incrementCounter,',
-          '// onPressed: _incrementCounter,',
-        );
-        source = source.replaceFirst(
-            '_counter++;',
-            '_counter++; throw "an error message";',
-          );
-        libMain.writeAsStringSync(source);
-
-        // Analyze in the current directory - no arguments
-        await runCommand(
-          command: AnalyzeCommand(workingDirectory: globals.fs.directory(projectPath)),
-          arguments: <String>['analyze', '--no-pub'],
-          statusTextContains: <String>[
-            'Analyzing',
-            'info $analyzerSeparator Avoid empty else statements',
-            'info $analyzerSeparator Avoid empty statements',
-            'info $analyzerSeparator The declaration \'_incrementCounter\' isn\'t',
-          ],
-          exitMessageContains: '3 issues found.',
-          toolExit: true,
-        );
-      });
+  tearDownAll(() {
+    tryToDelete(tempDir);
+  });
+
+  // Analyze in the current directory - no arguments
+  testUsingContext('working directory', () async {
```

Comment:
Can any of these be `testWithoutContext`?

Reply:
Unfortunately not, FlutterCommand itself accesses some context members so any command involving it will hit exceptions